### DISCUSSION TITLE
Compare files - ignore last row

### DISCUSSION THREAD

**Discussion by stpeters68** [WinMerge/winmerge#1659] [Q&A]
I want to compare files, ignoring the last line, since it is always different and I am only interested in differences in the range above it. This probably works with the line filter and regular expressions. Unfortunately I don't know how to do this and hope for your help.

I have attached a screenshot of the difference in the last line, which I would like to ignore.

![Screenshot 2023-01-09 132825](https://user-images.githubusercontent.com/122277834/211308237-d151d992-1841-4d11-8734-a236a264c972.jpg)

**Comment by sdottaka** [comment]
Unfortunately, WinMerge currently does not have the ability to ignore last-line-only differences.

However, if the last line always contains a particular string and no other line contains that string, you can ignore the difference in the last line by specifying the string as a line filter. .

The following is an example with a line filter that assumes that the string "Prophet Table Editor" always appears in the last line.

![image](https://user-images.githubusercontent.com/98126/211321140-1d7e22f3-1c9c-4456-beb8-457ecf246f6b.png)

**Comment by stpeters68** [comment]
Thanks for the help - I can handle the regular expressions now.

My next problem is that WinMerge recognizes some files as binary files and displays them as different. When I want to display the differences, the info comes up that they are identical.


![Screenshot 2023-01-11 114710](https://user-images.githubusercontent.com/122277834/211788754-f10d0f38-2226-469e-9292-a14a32dacdc4.png)
![Screenshot 2023-01-11 114831](https://user-images.githubusercontent.com/122277834/211788787-f6cc1d57-f816-4c3c-8b24-46eabfa4b925.png)

**Comment by sdottaka** [comment]
Unfortunately, the line filters are not applied to files detected as binary files.

Although this is a much more complicated and less preferred method, you can combine multiple Unpacker plug-ins to compare files with the last line stripped off, as shown below. You can then apply this combination of plug-ins to multiple pairs of files in the folder comparison window and re-compare them. 1.

1. select multiple files to compare in the folder comparison window, ignoring differences in the last line. 
2. right-click→Plugin Settings→Unpacker Settings→Select .... 
3. Enter `ReverseLines|SelectLines 2-|ReverseLines` in the Plugin Pipeline. 
4. Press F5 to compare again.

![image](https://user-images.githubusercontent.com/98126/212088755-7b4cd168-7e3b-448a-988b-8b73733a2e5d.png)
![image](https://user-images.githubusercontent.com/98126/212088897-1b121a1f-6565-4508-b128-2c827aebfc12.png)

**Comment by stpeters68** [comment]
I just had some time again to deal with the problem. Unfortunately, the proposed solution does not yet lead to the goal.

I have attached two files that are identical except for the last line.

When I compare them directly with the suggested line filter, they are shown to me as identical. If they are in different folders, I get the result "Files are different".
[File_in_Folder1.txt](https://github.com/WinMerge/winmerge/files/10455898/File_in_Folder1.txt)
[File_in_Folder2.txt](https://github.com/WinMerge/winmerge/files/10455900/File_in_Folder2.txt)

**Reply by sdottaka** [reply]
Have you tried the method from https://github.com/WinMerge/winmerge/discussions/1659#discussioncomment-4668188?

**Reply by stpeters68** [reply]
Yes I had - but tried again and realized that I had selected the folder and not the files in the folder.

Now it worked :-) Thanks for the support.
